File: e2e/tests/17-workspace-toolchains-multilang/fixtures/python-repo/todo.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class Todo:
    id: int
    title: str
    done: bool = False
    completed_at: Optional[str] = None


def add_todo(items: list[Todo], title: str) -> list[Todo]:
    next_id = len(items) + 1
    return [*items, Todo(id=next_id, title=title)]


def complete_todo(items: list[Todo], item_id: int, completed_at: str) -> bool:
    for item in items:
        if item.id == item_id:
            item.done = True
            item.completed_at = completed_at
            return True
    return False


def pending_count(items: list[Todo]) -> int:
    return len([item for item in items if not item.done])

File: e2e/tests/17-workspace-toolchains-multilang/fixtures/node-ts-repo/src/todo.ts
export type Todo = {
  id: number;
  title: string;
  done: boolean;
  completedAt?: string;
};

export function addTodo(list: Todo[], title: string): Todo[] {
  const item: Todo = {
    id: list.length + 1,
    title,
    done: false
  };
  return [...list, item];
}

export function completeTodo(list: Todo[], id: number, completedAt: string): boolean {
  for (const item of list) {
    if (item.id === id) {
      item.done = true;
      item.completedAt = completedAt;
      return true;
    }
  }
  return false;
}

export function pendingCount(list: Todo[]): number {
  return list.filter((item) => !item.done).length;
}
